lib/Runtime/Debug/DebugContext.h

```diff
@@ -51,11 +51,22 @@ namespace Js
         void SetHostDebugContext(HostDebugContext * hostDebugContext);
 
         void SetDebuggerMode(DebuggerMode mode);
+
         bool IsDebugContextInNonDebugMode() const { return this->debuggerMode == DebuggerMode::NotDebugging; }
         bool IsDebugContextInDebugMode() const { return this->debuggerMode == DebuggerMode::Debugging; }
         bool IsDebugContextInSourceRundownMode() const { return this->debuggerMode == DebuggerMode::SourceRundown; }
         bool IsDebugContextInSourceRundownOrDebugMode() const { return IsDebugContextInSourceRundownMode() || IsDebugContextInDebugMode(); }
 
+        bool IsDebuggerRecording() const { return this->IsDebugContextInDebugMode() && this->isDebuggerRecording; }
+        void SetIsDebuggerRecording(bool isDebuggerRecording)
+        {
+            if (this->IsDebugContextInDebugMode())
```

Comment:
Such an Assert would fail when `~AutoInitLibraryCodeScope` is run when not in `Debugging` mode. My thinking was that this check is necessary to protect against changing the state of `isDebuggerRecording` when not in `Debugging` mode. However, if we're saving and restoring the state anyway that might be unnecessary.